packages/number-input/src/NumberInput/NumberInput.spec.tsx

```diff
@@ -202,47 +206,64 @@ describe('packages/number-input', () => {
       expect((numberInput as HTMLInputElement).value).toBe('111');
     });
 
-    test('blur triggers onBlur callback', () => {
-      const { numberInput } = renderNumberInput({
-        label,
-        ...defaultProps,
-      });
-
-      userEvent.tab(); // focus
-      expect(numberInput).toHaveFocus();
-      userEvent.tab(); // blur
-
-      expect(defaultProps.onBlur).toHaveBeenCalledTimes(1);
-    });
-
     test('value changes when "up" arrow is clicked', () => {
-      const { container, numberInput } = renderNumberInput({
+      const { getByTestId, numberInput } = renderNumberInput({
         label,
         ...defaultProps,
       });
 
-      const upArrow = container.querySelector(
-        'button[aria-label="increment number"]',
-      );
+      const upArrow = getByTestId(arrowTestId.up);
 
       userEvent.click(upArrow as HTMLButtonElement);
       expect((numberInput as HTMLInputElement).value).toBe('1');
     });
 
     test('value changes when "down" arrow is clicked', () => {
-      const { container, numberInput } = renderNumberInput({
+      const { getByTestId, numberInput } = renderNumberInput({
         label,
         ...defaultProps,
       });
 
-      const upArrow = container.querySelector(
-        'button[aria-label="decrement number"]',
-      );
+      const downArrow = getByTestId(arrowTestId.down);
 
-      userEvent.click(upArrow as HTMLButtonElement);
+      userEvent.click(downArrow as HTMLButtonElement);
       expect((numberInput as HTMLInputElement).value).toBe('-1');
     });
 
+    describe('onBlur', () => {
+      test('callback triggers when focus leaves number input', () => {
+        const onBlur = jest.fn();
+        const { numberInput } = renderNumberInput({
+          label,
+          onBlur,
+          ...defaultProps,
+        });
+
+        userEvent.tab(); // focus
+        expect(numberInput).toHaveFocus();
+        userEvent.tab(); // blur
+
+        expect(onBlur).toHaveBeenCalledTimes(1);
+      });
+
+      test('callback triggers when focus leaves arrow buttons', () => {
```

Comment:
We should test the case(s) where `onBlur` should _not_ be called

Reply:
makes sense, added a test case